src/mad_spark_alt/core/provider_router.py

```
@@ -137,6 +222,89 @@ def _is_ollama_connection_error(self, error: Exception) -> bool:
+            try:
+                ip = ipaddress.ip_address(hostname)
+            except ValueError:
+                pass  # Not an IP address, hostname is fine
+            else:
+                # Successfully parsed as IP, check if it's private
+                if ip.is_private or ip.is_loopback or ip.is_link_local:
+                    raise ValueError(f"Private/internal IP not allowed: {hostname}")
+
+        # 3. Block cloud metadata endpoints (common SSRF targets)
+        if any(pattern in url for pattern in _BLOCKED_METADATA_PATTERNS):
+            raise ValueError(f"Cloud metadata endpoints not allowed: {url}")
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)